src/test/crimson/seastore/test_object_data_handler.cc

```diff
@@ -709,7 +743,77 @@ TEST_P(object_data_handler_test_t, random_overwrite) {
       logger().info("random_writes: {} done replaying/checking", i);
     }
     read(0, 4<<20);
-    unset_overwrite_threshold();
+    disable_delta_based_overwrite();
+  });
+}
+
+TEST_P(object_data_handler_test_t, overwrite_then_read_within_transaction) {
+  run_async([this] {
+    enable_delta_based_overwrite();
+    auto t = create_mutate_transaction();
+    auto base = 4096 * 4;
+    auto len = 4096 * 6;
+    write(*t, base, len, 'a');
+    submit_transaction(std::move(t));
+
+    t = create_mutate_transaction();
+    { 
+      auto pins = get_mappings(base, len);
+      assert(pins.size() == 1);
+      auto pin1 = remap_pin(*t, std::move(pins.front()), 4096, 8192);
+      auto ext = get_extent(*t, base + 4096, 4096 * 2);
+      ASSERT_TRUE(ext->is_exist_clean());
+      write(*t, base + 4096, 4096, 'b');
+      ext = get_extent(*t, base + 4096, 8192);
+      ASSERT_FALSE(ext->is_exist_mutation_pending());
```

Comment:
ASSERT_FALSE? Shouldn't it be testing exist_mutation_pending?

Reply:
That is on the to-do list (https://github.com/ceph/ceph/pull/54525#issuecomment-1892946060).
As you can see in the following lines, the current implementation only performs delta-over-write if` r->is_stable()` is true.
```
      for (auto &r : ops.to_remove) {
        if (r->is_stable() && !r->is_zero_reserved()) {
          pre_alloc_addr_removed.insert(r->get_key(), r->get_length());
        }
      }
```

Reply:
Found my comment https://github.com/ceph/ceph/pull/50419#issuecomment-1803305057, quote:

> I think the special point about EXIST_CLEAN is that its data is stable, but its mapping isn't. So probably we can introduce LBAMapping::is_stable() and LBAMapping::is_data_stable() if need to differenciate them explicitly.

I think that's a separate issue related to overwrite strategies which is quite flexible (unstable). Seems to me the test is possible to skip that part by using "ObjectDataBlock::overwrite()" directly, right?

Reply:
> ```
> for (auto &r : ops.to_remove) {
>   if (r->is_stable() && !r->is_zero_reserved()) {
>     pre_alloc_addr_removed.insert(r->get_key(), r->get_length());
>   }
> }
>```

It's easy to forget once the PRs are merged.

Let's take the chance to add TODO comments to the related logic to remember if the direction is reasonable to you, IIUC:
```
// TODO: Introduce LBAMapping::is_data_stable() to include EXIST_CLEAN extents
```

Reply:
Done.